ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTPayload.scala

```diff
@@ -262,6 +262,18 @@ object AuthServiceJWTCodec {
       )
   }
 
+  def readScopeBasedTokenFromString(
+      targetScope: String
+  )(value: String): Try[AuthServiceJWTPayload] =
+    for {
+      json <- Try(value.parseJson)
+      parsed <- Try(readScopeBasedToken(json))
+      validated <- parsed match {
+        case p: StandardJWTPayload if p.scope.contains(targetScope) => Success(p)
+        case _ => deserializationError(msg = "Unexpected scope in the token")
```

Comment:
looking at the type signature of `readScopeBasedToken`, you can theoretically have a custom token as well. I'd change that one to also return a `StandardJWTPayload` so this match becomes exhaustive then